kube-core/src/lib.rs

```diff
@@ -70,3 +70,6 @@ pub use error::ErrorResponse;
 
 mod version;
 pub use version::Version;
+
+pub mod error_boundary;
+pub use error_boundary::ErrorBoundary;
```

Comment:
as a more bikesheddy comment, the name here does kind of feel like it has something to do with bounds on the error, but to me this is more a conversion safety wrapper. Something like `Api<ConversionSafe<CaConfigMap>>` perhaps makes it a little clearer?

Reply:
I don't love either name, I was trying to evoke React's https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary (and replicated as Solid's https://docs.solidjs.com/concepts/control-flow/error-boundary).

Reply:
yeah, don't feel the react/solidjs name really translates that nicely to this, since we only really catch one error type;  it feels better to highlight the type of error we are guarding against imo.

maybe something along the lines
- {Deserialize,Conversion}Safe
- {Deserialize,Convert}Guard
- {Metadata,ObjectMeta}Fallback

(keeping the module name as `error_boundary` for housing such a struct / more specific structs is fine to me though)

Reply:
`DeserializeGuard` maybe? `DeserializeBoundary`? hmm

Reply:
slight preference for the first of those 2 personally

Reply:
Alright.

Reply:
9a0480b953db0a8ecc1120376e76effdb097fda7